### DISCUSSION TITLE
Remove postgres version selector

### DISCUSSION THREAD

**Discussion by phemmer** [grafana/grafana#38989] [Data Sources]
Currently postgres data source configuration offers a version selector. However [it contains](https://github.com/grafana/grafana/blob/6a39ac7/public/app/plugins/datasource/postgres/config_ctrl.ts#L83-L91) really old & unsupported postgres versions. https://www.postgresql.org/support/versioning/

As the oldest supported version is 9.6, one might argue that all versions prior to that should be removed. However once done, the version selector won't actually do anything. Right now the selector has 7 values, only 2 of them actually do something. The "9.4" version [controls the presence of ordered aggregates](https://github.com/grafana/grafana/blob/6a39ac7/public/app/plugins/datasource/postgres/query_ctrl.ts#L161-L170), and the "9.6" version [controls whether to offer timescaledb](https://github.com/grafana/grafana/blob/6a39ac7/public/app/plugins/datasource/postgres/config_ctrl.ts#L44-L50). So basically once the unsupported versions are removed, all versions will function exactly the same within Grafana.

In addition to general cleanup, these versions have been a source of confusion as people incorrectly assume that grafana doesn't work correctly if their version isn't listed (#24320, #31339). Granted #37166 did make this a little clearer. But having them still causes confusion as I don't know whether they're useless, or actually do something, and need to be worried about.

If at some point in the future, grafana does add functionality which depends on the version, then we can add it back, and with a less confusing implementation.

**Comment by gabor** [comment]
thank you for mentioning this, i converted this into an issue at https://github.com/grafana/grafana/issues/78393 .
